lib/semantics/tools.cc

```diff
@@ -503,6 +503,13 @@ bool IsAssumedLengthCharacterFunction(const Symbol &symbol) {
   return symbol.has<SubprogramDetails>() && IsAssumedLengthCharacter(symbol);
 }
 
+bool IsPolymorphic(const Symbol &symbol) {
+  if (const DeclTypeSpec * type{symbol.GetType()}) {
+    return type->IsPolymorphic();
+  }
+  return false;
```

Comment:
Can this be rewritten as type && type->IsPolymorphic() ?

Reply:
Yep.  But this section of code has been eliminated entirely in response to a comment from Tim.